### PR TITLE
Start "flexible" model_api and Subclasses

### PR DESCRIPTION
**Pull request by couasnonanais**
- First implementation for the flexible model_api
- beginning of subclass models. 
- We should discuss how to move next for model_api

### CODE REVIEW

File: hydromt/models/model_lumped.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        super().read()
+        # Other specifics to LumpedModel...
+
+    def write(self):
+        """Method to write the complete model schematization and configuration to file."""
+        super().write()
+    
+    def set_forcing_name():
+
+    def space_interpolation():
```
Comment (couasnonanais) [inline hydromt/models/model_lumped.py:47]: def setup_fromraster():
#read raster data
#zonal stats
#add to hydromt object

Comment (couasnonanais) [reply hydromt/models/model_lumped.py:47]: def setup_temporaldata(): #temporal raster and temporal vector
#Check soemthing for precip?
#In the lumped model always do the zonalstats

Comment (couasnonanais) [reply hydromt/models/model_lumped.py:47]: in setup_temporaldata: check resmapling in time and space to check model dimensions

File: hydromt/models/model_api.py [hunk 2]
```diff
@@ -640,7 +710,6 @@ def get_config(self, *args, fallback=None, abs_path=False):
         return value
 
     ## model parameter maps, geometries and spatial properties
-
     @property
     def staticmaps(self):
```
Comment (couasnonanais) [inline hydromt/models/model_api.py:714]: In the future, maybe have another proerty to store as auxiliary or metadata (not necessary to have smae reoslution)

File: hydromt/models/model_api.py [hunk 3]
```diff
@@ -844,17 +913,17 @@ def set_results(self, data, name=None, split_dataset=False):
     ## properties / methods below can be used directly in actual class
 
     @property
-    def crs(self):
-        """Returns coordinate reference system embedded in staticmaps."""
-        return self.staticmaps.raster.crs
+    def crs(self): 
+        """Returns coordinate reference system embedded in staticgeoms."""
+        return self.staticgeoms["region"].crs
```
Comment (couasnonanais) [inline hydromt/models/model_api.py:918]: Add if statements if crs is None (or a warning?)

File: hydromt/models/model_api.py [hunk 4]
```diff
@@ -844,17 +913,17 @@ def set_results(self, data, name=None, split_dataset=False):
...
 
     def set_crs(self, crs):
         """Embed coordinate reference system staticmaps metadata."""
         return self.staticmaps.raster.set_crs(crs)
 
     @property
-    def dims(self):
-        """Returns spatial dimension names of staticmaps."""
+    def dims(self): #TODO: Right now does not give an error but decide what to do with this
```
Comment (couasnonanais) [inline hydromt/models/model_api.py:925]: Add if statements for warning if staticmaps is empty

File: hydromt/models/model_api.py [hunk 5]
```diff
@@ -844,17 +913,17 @@ def set_results(self, data, name=None, split_dataset=False):
...
 
     @property
-    def crs(self):
-        """Returns coordinate reference system embedded in staticmaps."""
-        return self.staticmaps.raster.crs
+    def crs(self): 
+        """Returns coordinate reference system embedded in staticgeoms."""
+        return self.staticgeoms["region"].crs
 
     def set_crs(self, crs):
```
Comment (couasnonanais) [inline hydromt/models/model_api.py:920]: Remove and check
Or set to all objects

File: hydromt/models/model_api.py (hunk 4 above)
Comment (couasnonanais) [reply hydromt/models/model_api.py:925]: throw a Pending deprecation warning. For now check first if there is data (check xarray is empty) so doesn't bug

**Comment by couasnonanais** [comment]
@DirkEilander or @hboisgon : pull request ready to be reviewed. Could one of you give some feedback? Thanks!

**Review by DirkEilander — APPROVED** [review]

**Pull request merged**